src/chainstate/burn/operations/leader_block_commit.rs

```diff
@@ -442,6 +547,274 @@ mod tests {
+                                              network_id: BitcoinNetworkType::Mainnet,
+                                              bytes: Hash160([0; 20]) }
+                },
+            ],
+        });
+
+        let op = LeaderBlockCommitOp::parse_from_tx(16843019, &BurnchainHeaderHash([0; 32]), &tx).unwrap();
+
+        // should have 2 commit outputs, and a burn
+        assert_eq!(op.commit_outs.len(), 2);
+        assert_eq!(op.burn_fee, 50);
+
+        let tx = BurnchainTransaction::Bitcoin(BitcoinTransaction {
+            txid: Txid([0; 32]), vtxindex: 0,
+            opcode: Opcodes::LeaderBlockCommit as u8,
+            data: vec![1; 80],
+            inputs: vec![BitcoinTxInput {
+                keys: vec![],
+                num_required: 0,
+                in_type: BitcoinInputType::Standard
+            }],
+            outputs: vec![
+                BitcoinTxOutput {
+                    units: 10,
+                    address: BitcoinAddress { addrtype: BitcoinAddressType::PublicKeyHash,
+                                              network_id: BitcoinNetworkType::Mainnet,
+                                              bytes: Hash160([1; 20]) }
+                },
+                BitcoinTxOutput {
+                    units: 10,
+                    address: BitcoinAddress { addrtype: BitcoinAddressType::PublicKeyHash,
+                                              network_id: BitcoinNetworkType::Mainnet,
+                                              bytes: Hash160([2; 20]) }
+                },
+                BitcoinTxOutput {
+                    units: 29,
+                    address: BitcoinAddress { addrtype: BitcoinAddressType::PublicKeyHash,
+                                              network_id: BitcoinNetworkType::Mainnet,
+                                              bytes: Hash160([0; 20]) }
+                },
+            ],
+        });
+
+        // burn amount should have been 30, not 29
+        match LeaderBlockCommitOp::parse_from_tx(16843019, &BurnchainHeaderHash([0; 32]), &tx).unwrap_err() {
+            op_error::ParseError => {},
+            _ => unreachable!(),
+        };
+
+        let tx = BurnchainTransaction::Bitcoin(BitcoinTransaction {
+            txid: Txid([0; 32]), vtxindex: 0,
+            opcode: Opcodes::LeaderBlockCommit as u8,
+            data: vec![1; 80],
+            inputs: vec![BitcoinTxInput {
+                keys: vec![],
+                num_required: 0,
+                in_type: BitcoinInputType::Standard
+            }],
+            outputs: vec![
+                BitcoinTxOutput {
+                    units: 13,
+                    address: BitcoinAddress { addrtype: BitcoinAddressType::PublicKeyHash,
+                                              network_id: BitcoinNetworkType::Mainnet,
+                                              bytes: Hash160([1; 20]) }
+                },
+                BitcoinTxOutput {
+                    units: 13,
```

Comment:
I'm pretty sure this will work, but we will want to test that Bitcoin allows you to send BTC to the same address (i.e. create two UTXOs with the same scriptSig) in the same transaction.

Reply:
No; I think all block-commit transactions need to be the same size no matter what.